src/Microsoft.ML.Recommender/MatrixFactorizationTrainer.cs

```diff
@@ -331,9 +325,8 @@ internal static class Defaults
         /// <summary>
         /// The <see cref="TrainerInfo"/> contains general parameters for this trainer.
         /// </summary>
-        TrainerInfo ITrainer.Info => _info;
 
-        private readonly TrainerInfo _info;
+        public TrainerInfo Info { get; set; }
```

Comment:
This definitely shouldn't be publicly settable.

Also - what is the reason for this change?

Reply:
I will remove the `set` property

To implement `ITrainerEstimater` interface, see [this](https://github.com/dotnet/machinelearning/blob/edfd10f131abe97a7f7bb25c0070843a67b8b660/src/Microsoft.ML.Data/Training/ITrainerEstimator.cs#L12)

Reply:
To implement ITrainerEstimater you can use the same pattern as above - 

```
        /// <summary>
        /// The <see cref="TrainerInfo"/> contains general parameters for this trainer.
        /// </summary>
        TrainerInfo ITrainer.Info => _info;
```